crates/flipperzero/src/subghz/preset.rs

```diff
@@ -0,0 +1,64 @@
+use flipperzero_sys::{
+    FuriHalSubGhzPreset, FuriHalSubGhzPreset_FuriHalSubGhzPreset2FSKDev238Async,
+    FuriHalSubGhzPreset_FuriHalSubGhzPreset2FSKDev476Async,
+    FuriHalSubGhzPreset_FuriHalSubGhzPresetCustom,
+    FuriHalSubGhzPreset_FuriHalSubGhzPresetGFSK9_99KbAsync,
+    FuriHalSubGhzPreset_FuriHalSubGhzPresetIDLE,
+    FuriHalSubGhzPreset_FuriHalSubGhzPresetMSK99_97KbAsync,
+    FuriHalSubGhzPreset_FuriHalSubGhzPresetOok270Async,
+    FuriHalSubGhzPreset_FuriHalSubGhzPresetOok650Async,
+};
+
+#[derive(Clone, Debug)]
+pub enum SubGhzPreset<'a> {
+    /// default configuration
+    IDLE,
+    /// OOK, bandwidth 270kHz, asynchronous
+    Ook270Async,
+    /// OOK, bandwidth 650kHz, asynchronous
+    Ook650Async,
+    /// FM, deviation 2.380371 kHz, asynchronous
+    _2FSKDev238Async,
+    /// FM, deviation 47.60742 kHz, asynchronous
+    _2FSKDev476Async,
+    /// MSK, deviation 47.60742 kHz, 99.97Kb/s, asynchronous
+    MSK99_97KbAsync,
+    /// GFSK, deviation 19.042969 kHz, 9.996Kb/s, asynchronous
+    GFSK9_99KbAsync,
+    Custom(CustomSubGhzPreset<'a>),
+}
+
+impl<'a> SubGhzPreset<'a> {
+    pub fn into_furi_preset(&self) -> FuriHalSubGhzPreset {
+        match self {
+            SubGhzPreset::IDLE => FuriHalSubGhzPreset_FuriHalSubGhzPresetIDLE,
+            SubGhzPreset::Ook270Async => FuriHalSubGhzPreset_FuriHalSubGhzPresetOok270Async,
+            SubGhzPreset::Ook650Async => FuriHalSubGhzPreset_FuriHalSubGhzPresetOok650Async,
+            SubGhzPreset::_2FSKDev238Async => {
+                FuriHalSubGhzPreset_FuriHalSubGhzPreset2FSKDev238Async
+            }
+            SubGhzPreset::_2FSKDev476Async => {
+                FuriHalSubGhzPreset_FuriHalSubGhzPreset2FSKDev476Async
+            }
+            SubGhzPreset::MSK99_97KbAsync => FuriHalSubGhzPreset_FuriHalSubGhzPresetMSK99_97KbAsync,
+            SubGhzPreset::GFSK9_99KbAsync => FuriHalSubGhzPreset_FuriHalSubGhzPresetGFSK9_99KbAsync,
+            SubGhzPreset::Custom(_) => FuriHalSubGhzPreset_FuriHalSubGhzPresetCustom,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+/// This struct is defined for creating custom configurations. It enforces the null padding needed at the end. along with the power amplifier config?
+pub struct CustomSubGhzPreset<'a>(&'a [u8]);
+
+impl<'a> CustomSubGhzPreset<'a> {
+    /// # Safety
+    /// Read the manual for the CC1101 chip and review the [`furi_hal_subghz_load_custom_preset()`](https://github.com/flipperdevices/flipperzero-firmware/blob/0eaad8bf64f01a6f932647a9cda5475dd9ea1524/targets/f7/furi_hal/furi_hal_subghz.c#L162) function provided by the flipper API
```

Comment:
```suggestion
    /// Read the manual for the CC1101 chip and review the [`furi_hal_subghz_load_custom_preset()`].
    ///
    /// [`furi_hal_subghz_load_custom_preset()`]: (https://github.com/flipperdevices/flipperzero-firmware/blob/0eaad8bf64f01a6f932647a9cda5475dd9ea1524/targets/f7/furi_hal/furi_hal_subghz.c#L162) function provided by the flipper API
```

Reply:
Not quite sure what exactly the change is here. My assumption is for it to include both a link to the function in the `flipperzero_sys::furi_hal_etc` documentation as well as the link I provide to the C source code?

Reply:
My bad, I've accidentally trimmed the end of the comment sentence.

The intention here is to move the URL to the markdown footer so that the header text is easier to read